### PR TITLE
Improve documentation for threadpool

### PR DESCRIPTION
**Pull request by konsumlamm**

### CODE REVIEW

File: lib/pure/concurrency/threadpool.nim [hunk 1]
```diff
@@ -230,12 +231,12 @@ proc nimFlowVarSignal(fv: FlowVarBase) {.compilerproc.} =
...
-  ## to ``action``.
+  ## Blocks until `fv` is available and then passes its value
+  ## to `action`.
   ##
-  ## Note that due to Nim's parameter passing semantics this
-  ## means that ``T`` doesn't need to be copied so ``awaitAndThen`` can
-  ## sometimes be more efficient than `^ proc <#^,FlowVar[T]>`_.
+  ## Note that due to Nim's parameter passing semantics, this
+  ## means that `T` doesn't need to be copied, so `awaitAndThen` can
+  ## sometimes be more efficient than the `^ proc <#^,FlowVar[T]>`_.
```
Comment (timotheecour) [inline lib/pure/concurrency/threadpool.nim:239]: nit: in many of your PR's you're adding an article (`the`, `a`, etc) which IMO should be omitted when it refers to such kind of link. Maybe it's just a matter of taste.

**Review by timotheecour — APPROVED** [review]

File: lib/pure/concurrency/threadpool.nim (hunk 1 above)
Comment (konsumlamm) [reply lib/pure/concurrency/threadpool.nim:239]: My version is the grammatically correct version though, isn't it?

Comment (timotheecour) [reply lib/pure/concurrency/threadpool.nim:239]: how about (for future such cases, I don't really care about this isolated case) doing:

```nim
## sometimes be more efficient than the `^ proc <#^,FlowVar[T]>`_.
```
=>
```nim
## sometimes be more efficient than `^<#^,FlowVar[T]>`_.
```

it's still grammatically correct, shorter, and doesn't make a distinction between the fact that this symbol is a macro/template/proc/func (a user interested in knowing whether the linked symbol is a macro/template/proc/func can just click it, or, maybe docgen could automatically insert the symbol kind). Note that API's should often be allowed to switch between those (eg proc => func or proc => template) transparently, so distinguishing this in doc links shouldn't be needed.

Comment (konsumlamm) [reply lib/pure/concurrency/threadpool.nim:239]: Ok, I will keep that in mind.

**Pull request merged**